### ISSUE TITLE
Comparisons are not consistent across different batch sizes

### ISSUE THREAD

**Issue by rajatsen91** [cure-lab/LTSF-Linear#58]
The prediction and actual sizes [here](https://github.com/cure-lab/LTSF-Linear/blob/main/exp/exp_main.py#L303) are not the same across batch sizes. This means that the metrics calculated are not exactly comparable across different models trained with different batch sizes.

Therefore if my understanding is correct all models might need to be reevaluated. 

I think the culprit is here: https://github.com/cure-lab/LTSF-Linear/blob/main/data_provider/data_factory.py#L20

The drop_last should be False during the test evaluations.

**Comment by mixiancmx** [comment]
I think drop_last doesn't affect the comparison of models. As the influence can be ignored.
Models in our repo use batch size = 8, 16, 32. Take the case where the model trained with batch_size = 8 compared with the model trained with batch_size = 32 as an example. In the worst case, the model trained with batch_size 8 is evaluated on 24 test samples **more** than the one trained with batch_size 32.
1. 24 is a very small number compared with the size of the whole test set. For example, the test set of the dataset "weather" has more than 10,000 test samples.
2. Model has similar performance on those 24 test samples and the remaining samples. Suppose the test set is a list `test_list`, then test_list[-23:-1] is the difference. test_list[-24] and test_list[-1] have a large overlap. For example, when the input length is 336 and the output length is 720, one test sample has 1056 data points in total. **Test_list[-24] and test_list[-1] overlap 1032 data points** (1056-24). The influence of 24 data points is quite small and the performance of  test_list[-24] and test_list[-1] is similar.

**Comment by rajatsen91** [comment]
I agree that for larger test sets the metrics should not be affected by much. But for the smaller datasets like ILI, ETTH1&2 I found that there could be significant differences. 

This definitely does not change the ordering of methods in your paper (thanks for releasing the code as well) as the differences are too big. However for future work this will matter on the smaller datasets. I think this bug was corrected in the Autoformer repository. See the discussion here:  https://github.com/yuqinie98/PatchTST/issues/7

Thanks.
